### PR TITLE
Fix general merkle keypath to start w/ last op's key

### PR DESCRIPTION
**Pull request by jaekwon**
* [ ] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [x] Wrote tests
* [x] Updated CHANGELOG_PENDING.md

General Merkle ProofOperator was iterating over the keys in the wrong order.  The keypath should start with the last operation to run (e.g. the top-level root hash operation), not end with it.

NOTE: This is related to https://github.com/cosmos/cosmos-sdk/pull/2633/files

### CODE REVIEW

File: crypto/merkle/proof.go [hunk 1]
```diff
@@ -43,18 +43,19 @@ func (poz ProofOperators) Verify(root []byte, keypath string, args [][]byte) (er
 	for i, op := range poz {
 		key := op.GetKey()
 		if len(key) != 0 {
-			if !bytes.Equal(keys[0], key) {
-				return cmn.NewError("Key mismatch on operation #%d: expected %+v but %+v", i, []byte(keys[0]), []byte(key))
+			lastKey := keys[len(keys)-1]
```
Comment (melekes) [inline crypto/merkle/proof.go]: should we check that `len(keys) > 0` especially when we're modifying them later `keys = keys[:len(keys)-1]`

**Comment by ebuchman** [comment]
Need to update docs in 

- [ ] https://github.com/tendermint/tendermint/blob/develop/docs/spec/abci/abci.md#proof
- [ ] https://github.com/tendermint/tendermint/blob/develop/docs/spec/abci/apps.md#query-proofs

File: crypto/merkle/proof.go (hunk 1 above)
Comment (ebuchman) [reply crypto/merkle/proof.go]: KeyPathToKeys ensures that the keys aren't empty, unless there's an error

File: crypto/merkle/proof_test.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+	// prt := NewProofRuntime()
+	// prt.RegisterOpDecoder(ProofOpDomino, DominoOpDecoder)
+
+	// ProofOperators setup
+	op1 := NewDominoOp("KEY1", "INPUT1", "INPUT2")
+	op2 := NewDominoOp("KEY2", "INPUT2", "INPUT3")
+	op3 := NewDominoOp("", "INPUT3", "INPUT4")
+	op4 := NewDominoOp("KEY4", "INPUT4", "OUTPUT4")
+
+	// Good
```
Comment (ebuchman) [inline crypto/merkle/proof_test.go]: would make a good table driven test

**Comment by ebuchman** [comment]
Oh nvm, we haven't even written the keypath spec ...

**Pull request merged**